code/modules/mob/dead/observer/observer.dm

```diff
@@ -505,7 +505,53 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	var/turf/loc = get_turf(src)
 	show_air_status_to(loc, usr)
 
-/mob/dead/observer/verb/follow()
+
+//Returns a list of all mobs with their name
+/proc/getmobs(ghostfollow = FALSE)
+
+	var/list/mobs = sortmobs()
+	var/list/names = list()
+	var/list/creatures = list()
+	var/list/namecounts = list()
+	for(var/mob/M in mobs)
+		if(isobserver(M) && ghostfollow && M.client?.holder && M.client.holder.fakekey /*&& M.is_preference_enabled(/datum/client_preference/holder/stealth_ghost_mode)*/)
+			continue
+		var/name = M.name
+		if (name in names)
+			namecounts[name]++
+			name = "[name] ([namecounts[name]])"
+		else
+			names.Add(name)
+			namecounts[name] = 1
+		if (M.real_name && M.real_name != M.name)
+			name += " \[[M.real_name]\]"
+		if (M.stat == 2)
+			if(istype(M, /mob/dead/observer))
+				name += " \[ghost\]"
+			else
+				name += " \[dead\]"
+		creatures[name] = M
+
+	return creatures
+
+/proc/getmobs_ghost_follow()
+	return getmobs(TRUE)
+
+/mob/dead/observer/verb/follow(input in getmobs_ghost_follow())
+	set category = "Ghost"
+	set name = "Follow" // "Haunt"
+	set desc = "Follow and haunt a mob."
+
+	if(!input)
+		input = input(usr, "Select a mob:", "Ghost Follow") as null|anything in getmobs_ghost_follow()
+	if(!input)
+		return
+
+	var/target = getmobs_ghost_follow()[input]
```

Comment:
Is this necessary?  `getmobs_ghost_follow` seems kind of expensive, and this means it's called three times:
- when fetching the arguments for the verb
- when running the verb without an argument
- when getting `target` from the name of the mob.
It's *probably* fine as-is, but ideally we'd have it be argument-less and cache the result of `getmobs_ghost_follow()` instead.

Reply:
Maybe not;  I snagged all three of these from citcode while kinda head empty no thoughts.

I'll go though the logic and consider it.

Also, thank you for the review!

Reply:
Considering it a bit more, I think the fact it's a user-triggered verb (meaning calls are relatively rare) makes it an acceptable tradeoff.  Having getmobs_ghost_follow in the argument list is required for autocompletion in the text bar, and I do like that and use it.  I don't think there's a good way to optimize it while keeping that behavior.